flask/app/variants.py

```diff
@@ -0,0 +1,424 @@
+                .contains_eager(models.Genotype.variant)
+                .contains_eager(models.Variant.gene),
+                contains_eager(models.Dataset.tissue_sample)
+                .contains_eager(models.TissueSample.participant)
+                .contains_eager(models.Participant.family),
+            )
+            .filter(or_(*filters), models.users_groups_table.columns.user_id == user_id)
+        )
+
+    else:
+        app.logger.debug("Processing query - unrestricted based on user id.")
+
+        query = (
+            db.session.query(models.Dataset)
+            .join(models.Participant.tissue_samples)
+            .join(models.TissueSample.datasets)
+            .join(models.Dataset.genotype)
+            .join(models.Genotype.analysis, models.Genotype.variant)
+            .join(models.Variant.gene)
+            .join(models.Participant.family)
+            .options(
+                contains_eager(models.Dataset.genotype)
+                .contains_eager(models.Genotype.variant)
+                .contains_eager(models.Variant.gene),
+                contains_eager(models.Dataset.tissue_sample)
+                .contains_eager(models.TissueSample.participant)
+                .contains_eager(models.Participant.family),
+            )
+            .filter(or_(*filters))
+        )
+
+    app.logger.info(request.accept_mimetypes)
+
+    if request.accept_mimetypes["application/json"]:
+
+        app.logger.info("Defaulting to json response")
+
+        q = query.all()
+        app.logger.info(
+            [
+                {
+                    "participant_codename": dataset.tissue_sample.participant.participant_codename,
+                    "family_codename": dataset.tissue_sample.participant.family.family_codename,
+                }
+                for dataset in q
+            ]
+        )
+        return jsonify(
+            [
+                {
+                    "participant_codename": dataset.tissue_sample.participant.participant_codename,
+                    "family_codename": dataset.tissue_sample.participant.family.family_codename,
+                    "dataset": [
+                        {
+                            "variants": [
+                                {
+                                    **asdict(genotype),
+                                    **asdict(genotype.variant),
+                                    "gene": genotype.variant.gene.hgnc_gene_name,
+                                }
+                                for genotype in dataset.genotype
+                            ],
+                        }
+                    ],
+                }
+                for dataset in q
+            ]
+        )
+    elif request.accept_mimetypes["text/csv"]:
+
+        app.logger.info("text/csv Accept header requested")
+
+        try:
+            sql_df = pd.read_sql(query.statement, query.session.bind)
+        except:
+            abort(500, "Unexpected error")
```

Comment:
In these cases we should really try to log the exception (`app.logger.error("message", exc_info=True)`) in case